File: src/main/java/io/github/takenoha/towerdefense/paper/ResourceVaultInventoryHolder.java
package io.github.takenoha.towerdefense.paper;

import java.util.Objects;
import java.util.UUID;
import org.bukkit.inventory.Inventory;
import org.bukkit.inventory.InventoryHolder;

/** Identifies the read-only team point wallet screen opened from a core. */
public final class ResourceVaultInventoryHolder implements InventoryHolder {
    private final UUID coreId;
    private Inventory inventory;

    public ResourceVaultInventoryHolder(UUID coreId) {
        this.coreId = Objects.requireNonNull(coreId, "coreId");
    }

    public UUID coreId() {
        return coreId;
    }

    void attach(Inventory inventory) {
        this.inventory = Objects.requireNonNull(inventory, "inventory");
    }

    @Override
    public Inventory getInventory() {
        if (inventory == null) {
            throw new IllegalStateException("the GUI inventory has not been attached");
        }
        return inventory;
    }
}

File: src/main/kotlin/io/github/takenoha/towerdefense/paper/ResourceVaultInventoryHolder.kt
package io.github.takenoha.towerdefense.paper

import java.util.Objects
import java.util.UUID
import org.bukkit.inventory.Inventory
import org.bukkit.inventory.InventoryHolder

/** Identifies the read-only team point wallet screen opened from a core. */
class ResourceVaultInventoryHolder(coreId: UUID) : InventoryHolder {
    private val coreIdValue: UUID = Objects.requireNonNull(coreId, "coreId")
    private var inventory: Inventory? = null

    fun coreId(): UUID = coreIdValue

    fun attach(inventory: Inventory) {
        this.inventory = Objects.requireNonNull(inventory, "inventory")
    }

    override fun getInventory(): Inventory =
        inventory ?: throw IllegalStateException("the GUI inventory has not been attached")
}
